src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -287,6 +265,29 @@ void GinIndexStore::cancel() noexcept
         postings_file_stream->cancel();
 }
 
+void GinIndexStore::initSegmentId()
+{
+    String segment_id_file_name = getName() + GIN_SEGMENT_ID_FILE_TYPE;
+
+    UInt32 result;
+    if (storage->existsFile(segment_id_file_name))
+    {
+        std::unique_ptr<DB::ReadBufferFromFileBase> istr = this->storage->readFile(segment_id_file_name, {}, std::nullopt, std::nullopt);
+
+        uint8_t version = 0;
+        readBinary(version, *istr);
+
+        if (version > static_cast<std::underlying_type_t<Format>>(CURRENT_GIN_FILE_FORMAT_VERSION))
+            throw Exception(ErrorCodes::UNKNOWN_FORMAT_VERSION, "Unsupported full-text index version {}", version);
```

Comment:
Could we also print the current GIN file format version, e.g.

```cpp
throw Exception(ErrorCodes::UNKNOWN_FORMAT_VERSION, "Only text indexes with version {} are supported, found index has version {}", version, static_cast<std::underlying_type_t<Format>>(CURRENT_GIN_FILE_FORMAT_VERSION));
```

(also in l. 212)